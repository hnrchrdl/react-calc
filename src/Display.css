.display {
  width: 100%;
  border: 1px solid salmon;
  margin: 10px auto;
  box-sizing: border-box;
  height: 50px;
  line-height: 50px;
}
