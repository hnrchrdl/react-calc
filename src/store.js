import { createStore, combineReducers } from 'redux';
import calcReducer from './calculator';

export default createStore(combineReducers({ calculator: calcReducer }));
