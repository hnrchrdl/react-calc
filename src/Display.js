import React from 'react';
import './Display.css';
import { connect } from 'react-redux';

const Display = ({ result, steps }) => {
  return (
    <div className="display">
      {result ? (
        <span>{result}</span>
      ) : (
        (steps || []).map((step, idx) => <span key={idx}>{step}</span>)
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  steps: state.calculator.steps,
  result: state.calculator.result
});
const mapDispatchToProps = dispatch => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Display);
