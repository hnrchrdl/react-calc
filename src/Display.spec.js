import React from 'react';
import renderer from 'react-test-renderer';
import Display from './Display';
import store from './store';

it('matches snapshots', () => {
  const component = renderer.create(<Display store={store} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
