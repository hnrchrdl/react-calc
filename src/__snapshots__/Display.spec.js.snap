// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="display"
/>
`;
